third_party/forked/golang/reflect/deep_equal.go

```diff
@@ -112,10 +114,17 @@ func (e Equalities) deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool,
 	if fv, ok := e[v1.Type()]; ok {
 		return fv.Call([]reflect.Value{v1, v2})[0].Bool()
 	}
-	if v1.CanAddr() {
-		if fv, ok := e[v1.Addr().Type()]; ok {
+
+	if fv, ok := e[reflect.PtrTo(v1.Type())]; ok {
```

Comment:
paging @deads2k for eyes on reflect code

Reply:
It would be really great if the only changes to the forked code were hooks that were then used by the `semantic` package.  It's really unclear how much of this file is upstream and how much is ours.  I don;t want to review upstream code.

Reply:
done with extra commit.